test/karma-test-shim.js

```diff
@@ -87,9 +87,38 @@ function configureTestBed() {
 
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
 
-    testing.TestBed.initTestEnvironment(
+    var testBed = testing.TestBed.initTestEnvironment(
       testingBrowser.BrowserDynamicTestingModule,
       testingBrowser.platformBrowserDynamicTesting()
     );
+
+    applyAngularTestingPatches(testBed);
+  });
+}
+
+/** Applies patches and workarounds to Angular's testing package. */
+function applyAngularTestingPatches(testBed) {
+  // Original resetTestingModule function of the TestBed.
+  var _resetTestingModule = testBed.resetTestingModule;
+
```

Comment:
Maybe add something like...
```js
if (!testBed._activeFixtures) {
  throw Error('The monkey-patch for TestBed.resetTestingModule has failed ' +
   'because a new version of Angular has removed the _activeFixtures member. ' +
   'This needs to be fixed in karma-test-shim.js');
}
```